### ISSUE TITLE
Potential bug in BaseCurrency

### ISSUE THREAD

**Issue by emperorstarfinder** [Virtual-Universe/Virtual-Universe#18]
User reports the following in Issue #16 

[LLClient View]: Caught exception while processing OpenMetaverse.Packets.GroupAccountSummaryRequestPacket for Mr ReaL_Life, Object reference not set to an instance of an object. at Universe.Modules.Currency.BaseCurrencyServiceModule.Balance(UUID agentID) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Modules\Currency\BaseCurrency\Base.CurrencyServices.cs:line 221
at Universe.Modules.Avatar.Groups.GroupMoneyModule.client_OnGroupAccountSummaryRequest(IClientAPI client, UUID agentID, UUID groupID, UUID requestID, Int32 currentInterval, Int32 intervalDays) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Modules\Avatar\Groups\GroupMoneyModule.cs:line 181
at Universe.ClientStack.LLClientView.HandleGroupAccountSummaryRequest(IClientAPI client, Packet Packet) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\ClientStack\LLClientView.cs:line 11270
at Universe.ClientStack.LLClientView.ProcessSpecificPacketAsync(Object state) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\ClientStack\LLClientView.cs:line 729

**Comment by LifeGold** [comment]
When about expected Version 1.0.2 RC2?

**Comment by emperorstarfinder** [comment]
Version 1.0.2 RC1 was just released last week we don't usually release the RC2 for at least 1 to two months to give us time to be sure bugs introduced in RC1 are corrected.

**Comment by emperorstarfinder** [comment]
Further testing confirms this issue is related to the new incomplete implementation of the ProductInfo() features relating to the new Homes product service being implemented.  This error is known to us and is being worked on with the completion of its implementation potentially ready prior to Version 1.0.2 RC2 being released.  There is no further need to report this particular error message to us until after we have completed implementation of this matter.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
The issue is not really an error, and it is clear why. And not even the fact that it will be fixed in version RC2 with S. but what about those who makes the server not the day for 3 friends. Now on the server no work impossible to undertake, because the loading of textures and everything, and to turn off this module does not work. And you say that version 2 will be a month or two. Who makes the server, I can only sit and wait for the version with the RC2. May have the opportunity to version 2, to return the module to have the ability to build locations and do everything else on the server, and from version RC2 to remove this module. To save time those who are preparing the servers to run on new versions.

**Comment by emperorstarfinder** [comment]
The currency still works even with this error.  This error just relates to ProductInfo() which does not affect uploading textures, creating groups, etc.  The rest of the currency services do work properly.  We do not expect it to take us a month to push the fix for ProductInfo() to master as this will be an immediate correction.  There is no need to turn the module off as the error does not affect any current services or caps required by the viewer.

Why do we release RC2 at least one to two months after RC1?  The answer is that it gives us time to make any final tweaks or correct any bugs we receive reports on in RC1.  Remember a RC (Release Candidate) is still considered a beta version which is still moving toward a final release for that particular version.  1 to 2 months is a standard bare minimum in the development process.  When there is a critical bug between the time RC1 is released up until RC2 is released discovered we will always push a fix for this as quickly as we can once we have identified the fix and have corrected it and have proven to our satisfaction the bug has been corrected.  

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
What could be the problem that does not show balance and loads of texture. Downloaded new server was installed on a clean database with the same result.

**Comment by emperorstarfinder** [comment]
When you say the balance isnt shown do you mean in the viewer or the builtin webui?  

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Add texture error

Mar 11 11:47:46 -  [BASE HTTP SERVER]: HandleRequest() threw System.NullReferenceException: Object reference not set to an instance of an object.
   at Universe.Modules.Currency.BaseCurrencyServiceModule.get_Config() in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Modules\Currency\BaseCurrency\Base.CurrencyServices.cs:line 47
   at Universe.Modules.Currency.BaseCurrencyServiceModule.get_UploadCharge() in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Modules\Currency\BaseCurrency\Base.CurrencyServices.cs:line 198
   at Universe.Services.InventoryCAPS.ChargeUser(String asset_type, OSDMap map, Int32& charge, Int32& resourceCost) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Services\GenericServices\CapsService\CAPModules\InventoryCAPS.cs:line 338
   at Universe.Services.InventoryCAPS.ChargeUser(String asset_type, OSDMap map) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Services\GenericServices\CapsService\CAPModules\InventoryCAPS.cs:line 324
   at Universe.Services.InventoryCAPS.NewAgentInventoryRequest(String path, Stream request, OSHttpRequest httpRequest, OSHttpResponse httpResponse) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Services\GenericServices\CapsService\CAPModules\InventoryCAPS.cs:line 289
   at Universe.Framework.Servers.HttpServer.GenericStreamHandler.Handle(String path, Stream request, OSHttpRequest httpRequest, OSHttpResponse httpResponse) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Framework\Servers\HttpServer\Http Methods\GenericStreamHandler.cs:line 51
   at Universe.Framework.Servers.HttpServer.BaseHttpServer.HandleRequest(HttpListenerContext context) in c:\Users\Administrator\Desktop\Virtual-Universe-master\Universe\Framework\Servers\HttpServer\BaseHttpServer.cs:line 385

**Comment by emperorstarfinder** [comment]
i assume this error just started in the last day or two?

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Once released RC1 and I immediately asked when is RC2 with the thought that you have disabled it. If not for that I would have never bothered you and sat waiting when will RC2

**Comment by emperorstarfinder** [comment]
Oh I see the confusion now,

Noah Indicated he would suppress the error relating to productInfo() until that got implemented but he didn't mention shutting off currency.  

The error you just posted does look like a new error which we will have to test to determine whats going on.  I will add to my list of things to investigate when i get home from work tonight in a couple of hours in case Noah doesn't see this and fix it before i get home.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Thank you

**Comment by LifeGold** [comment]
- Some corrections
- Some general cleanup

Now the error with all scripts on the server 

Mar 12 02:24:02 -  [ScriptData]: Couldn't start script PoseBall, 5551966f-6904-4d9b-bb04-4aefd64822fb at <88.65867, 76.11019, 25.20796> in Welcome Center since asset ID 83dd3072-d08d-47bd-a6f5-94bfa04d785c could not be found

**Comment by emperorstarfinder** [comment]
That would be because your current dlls in the Data/ScriptEngine directory contain configuration for DotNetEngine.  Just delete those dlls in that folder and restart your server it should fix that and create new dlls for each script.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by emperorstarfinder** [comment]
After some preliminary testing tonight, it would appear the upload texture error with currency might have been caused by the configuration files.  The cleanup appears it has corrected it as I am not seeing that error at the moment.  If that particular error persists please let us know what you were doing when the error occurs, what texture where you attempting to upload, was it part of an OAR?, and provide a copy of the texture and logs will help.  We will keep that on our list for references.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
It's happening again, just loaded the picture is not the oar and IAR files, and just using the texture loader.

**Comment by emperorstarfinder** [comment]
Hmmmm,

I will need to look into this more then as I am not able to reproduce this error.  In the meantime we can rule out easily whether its something with the configuration files, texture itself, or just some anomaly on Windows Server 2012 by having you do as follows:

Note: I will assume you use Windows 7, Windows 8.1, or Windows 10  Though the process would be the same for linux except the exe files i mention would be the sh files.

1 Install MySQL Database Server (Preferably MySQL 5.5 and workbench.

2. Using MySQL Workbench create the database using the info in UniverseDocs/Create Database Tables.txt.

3. be sure your firewall ports are unblocked on windows and in your modem/ router firewall as this will cause problems for networking. (Ports I unblock are 8000-8999 TCP/UDP and 9000-9999 TCP/UDP for both Incoming and Outgoing and you want to be sure to select Allow Connection for outgoing_.

4. Run Universe.Server.exe let it set itself up initially.  Once its ready run the configuration by typing run configurator and hitting enter.  Answer the questions it asks for configuration.

5. Once it has completed reconfiguring it will ask you to please restart for configuration to take effect so do so.

6. After relaunching Universe.Server.exe run the command create user and enter the info it will ask you for.  This will create a new regular user.  Do not use the system users for anything other then what they are intended for.

7. Next create an estate owned by the user you just created in Step 6.  The command is create estate.  It will ask for the estate name and who the estate owner will be.  You can name the estate anything you like and the owner will be the user you just created.

8. Once you have that ready launch Universe.exe and let it do its thing.  It will ask you a few questions about how you want to configure your region.  (For the purposes of testing just use the defaults and give the region any name you like).

9. Once Universe.exe is completed startup we want to do the run configurator command to configure grid mode.  It will ask you a few questions so answer these questions and restart Universe.exe when it completes configuration and asks you.

10. Upon restarting Universe.exe with its new configuration it will ask you for the name of the estate owner for your region.  This is the name of the new user we created in Universe.Server.exe so give that name and it will join the region to that estate.

11. In Config/Grid/ServerConfiguration/GridInfoService.ini (which Universe.Server.exe has configured during configuration process you need to add the following lines:

SendGridInfoToViewerOnLogin = true
CurrencySymbol = "VU$"
RealCurrencySymbol = "$"

and restart Universe.Server.exe for this to take effect.

Do not touch any other configuration file as we have now configured what we need.

12. Log in on the user we created and try uploading a texture.  If the error dissappears and does not happen when uploading the texture then we will know that its not the texture but likely just an issue in the configuration or Windows Server 2012.  If it still happens when following these configuration steps then we will know something is wrong and I will want to see the logs of it so I can look at what exactly is going on along with the texture your trying to upload.

Remember when doing this you will see the error about ProductInfo() which we already know about that error and are in process of correcting that.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Alas just as well, the only thing that saves the old installation folder to the bin which was before the release of RC1 it works with your favorite configuration files

**Comment by emperorstarfinder** [comment]
SO i assume following the configuration steps I outline works correctly?

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Changes in fact present, as before.

**Comment by emperorstarfinder** [comment]
Then i will have to do some digging.  it could be that it is relating to something in the way windows is configured for the area you live in as well that is causing it though doubtful.  If you followed the steps i outlined it should work correctly.

it might take a hour or two before ill know what is going on if there is some strange happening in he code that we dont know about.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Thank you I will wait for a response

**Comment by emperorstarfinder** [comment]
Also when you tested locally on your pc (not your server) what operating system did you use?

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Windows 8 32bit

**Comment by emperorstarfinder** [comment]
Windows 8 or 8.1?

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Sorry 8.1

**Comment by emperorstarfinder** [comment]
hmmmm that might explain the problem then if that is the case if your able to perhaps test it on Windows 7 or Windows 10 following the steps i outlined.  it is possible that error is just some strange thing going on in windows 8.1 Most people are upgrading from Windows 8.1 to Windows 10

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
But where to find 10 or 7 I have on both tablets and 8.1((((

**Comment by emperorstarfinder** [comment]
Windows 8.1 you should get an upgrade message for win 10  you can also get a iso image for windows 7 from getintopc.com which you can use in virtual box or VMWare  which is what i do when i need to do a fast test of something for another OS while developing

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by emperorstarfinder** [comment]
And just out of curiosity  have you tried this with more then one image your uploading (texture) and gotten the same error and we are talking about just a simple picture and not a mesh object of some sort.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
What would you advise to rent a server on Ubuntu 14.04 LTS (64 bit), Debian 8 (64 bit), CentOS 7 (64 bit)? And how do I then compile the server your under this operating?

**Comment by emperorstarfinder** [comment]
Okay as far as recommending a linux server distribution, I generally run my grid on Ubuntu (Linden Labs runs theirs on Debian last i knew).  Debian and Ubuntu are pretty similar in installing packages etc.

As for compiling the difference will be youll need to install a few mono packages which you can find the url to the instructions how to do that on the main README.md file for this repository.  You also will after want to install monodevelop which is like visual studio.  thats easy to do on ubuntu with the sudo apt-get install monodevelop command in terminal.

To build the architecture instead of using runprebuild.bat youll use runprebuild.sh

compiling the code you can check: https://github.com/Virtual-Universe/Virtual-Universe/blob/master/VirtualUniverse/UniverseDocs/Build%20Your%20Own.txt which will give you the information you need for compiling.

to run the servers youll use the sh files instead of the exe files.  (Universe.Server.sh if running in grid mode and Universe.sh)

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Thank you, and the wither with this window

**Comment by emperorstarfinder** [comment]
Just be sure that if you do go linux that you don't edit any configuration files let the built in configurator in the virtual universe consoles edit the necessary files for you just follow the steps i provided in this issue earlier it should work properly.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by LifeGold** [comment]
Thanks again I will try to put. Here I think there is enough space, or else to take then not to put on a new server((( don't you think? http://prntscr.com/aehe6v

**Comment by emperorstarfinder** [comment]
that should be enough to launch your grid eventually youll have to upgrade the space but it should be a good start

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by emperorstarfinder** [comment]
These issues should now be fixed with the updates to the CAPS service

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team

**Comment by emperorstarfinder** [comment]
Errors appear to be fixed at this time with last nights commits.  Should further issues arise a new issue can be opened using the template provided for reporting issues.

Emperor Starfinder
CEO/ Founder
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Second Galaxy Development Team
